From e7f941ebde84805112bf0cba774f6ce885e147fc Mon Sep 17 00:00:00 2001
From: Wei Yang <richard.weiyang@gmail.com>
Date: Sat, 17 Jun 2017 09:17:37 +0200
Subject: [PATCH] include/linux/mmzone.h: simplify zone_intersects()
Patch-mainline: yes as a part of f1dd2cd13c4b
References: bnc#1047506

mhocko@suse.cz:
Please note that this follow up fix has been folded to the final patch merged as
f1dd2cd13c4b ("mm, memory_hotplug: do not associate hotadded memory to zones until
online"). It turned out it is not just a cleanup as originally thought and it fixes
a real bug.

To make sure a range intersects a zone, only two comparisons are
necessary.  Simplify the function a little.

Link: http://lkml.kernel.org/r/20170616092335.5177-1-richard.weiyang@gmail.com
Signed-off-by: Wei Yang <richard.weiyang@gmail.com>
Cc: Michal Hocko <mhocko@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Acked-by: Michal Hocko <mhocko@suse.cz>

---
 include/linux/mmzone.h |   10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

--- a/include/linux/mmzone.h
+++ b/include/linux/mmzone.h
@@ -582,11 +582,11 @@ static inline bool zone_intersects(struc
 {
 	if (zone_is_empty(zone))
 		return false;
-	if (zone->zone_start_pfn <= start_pfn && start_pfn < zone_end_pfn(zone))
-		return true;
-	if (start_pfn + nr_pages > zone->zone_start_pfn)
-		return true;
-	return false;
+	if (start_pfn >= zone_end_pfn(zone) ||
+	    start_pfn + nr_pages <= zone->zone_start_pfn)
+ 		return false;
+
+	return true;
 }
 
 /*
